### PR TITLE
Dev options

### LINKED ISSUE #569: Add Developer options screen
1) Add "Developer options" as first button in the list in Settings screen - use [Data Object](https://fonts.google.com/icons?selected=Material%20Symbols%20Outlined%3Adata_object%3AFILL%400%3Bwght%40400%3BGRAD%400%3Bopsz%4048) material icon and style should match specs below;
2) Move all the current buttons from Settings screen to "Developer options" screen, except the "Logout" button;
3) Change "Logout" button text to "Sign-out", use [Logout](https://fonts.google.com/icons?selected=Material%20Symbols%20Outlined%3Alogout%3AFILL%400%3Bwght%40400%3BGRAD%400%3Bopsz%4048) material icon, change style to match specs below;
4) No need to add "Downloads" or "About" chips as scope for this ticket;

|Redlines|Sample|
|---|---|
|Screen Shot 2022-09-12 at 13 37 29|Screen Shot 2022-09-12 at 13 31 49|

### PR DESCRIPTION
**Pull request by laiyichin**
#### WHAT
Add Developer options screen for Issue #569

#### WHY
Better organisation for developer options.

Fixes #569

#### HOW
- Add extra DeveloperOptionsScreenViewModel and DeveloperOptionsScreen
- Move all options (except Logout) to the newly created DeveloperOptions class
- Modify the ActionSetting to match the redlines

#### Checklist :clipboard:
- [N/A] Add explicit visibility modifier and explicit return types for public declarations
- [X] Run spotless check
- [X] Run tests
- [N/A] Update metalava's signature text files

dev_options

### CODE REVIEW

**Comment by yschimke** [comment]
Can you merge changes in from main to resolve the conflict.

File: media-sample/src/main/java/com/google/android/horologist/mediasample/ui/settings/DeveloperOptionsScreen.kt [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+import com.google.android.horologist.mediasample.domain.proto.SettingsProto.OffloadMode
+import com.google.android.horologist.mediasample.ui.navigation.navigateToAudioDebug
+import com.google.android.horologist.mediasample.ui.navigation.navigateToSamples
+
+@Composable
+fun DeveloperOptionsScreen(
+    focusRequester: FocusRequester,
+    state: ScalingLazyListState,
+    developerOptionsScreenViewModel: DeveloperOptionsScreenViewModel,
+    modifier: Modifier = Modifier,
```
Comment (luizgrp) [inline media-sample/src/main/java/com/google/android/horologist/mediasample/ui/settings/DeveloperOptionsScreen.kt:48]: there is a [convention / rule](https://developer.android.com/reference/kotlin/androidx/compose/ui/Modifier) in Compose that the modifier parameter:

> should appear as the first optional parameter in the parameter list; after all required parameters (except for trailing lambda parameters) but before any other parameters with default values.

**Review by luizgrp — APPROVED** [review]
LGTM 👍 

it would be nice if you could attach a screenshot of the screen to the PR description so we can see the result when reviewing it!

File: media-sample/src/main/java/com/google/android/horologist/mediasample/ui/settings/DeveloperOptionsScreen.kt (hunk 1 above)
Comment (laiyichin) [reply media-sample/src/main/java/com/google/android/horologist/mediasample/ui/settings/DeveloperOptionsScreen.kt:48]: iiuc. Moved modifier to last.

File: media-sample/src/main/java/com/google/android/horologist/mediasample/ui/settings/DeveloperOptionsScreenViewModel.kt [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+    fun showDialog(message: String) {
+        snackbarManager.showMessage(
+            UiMessage(
+                message = message,
+                error = true
+            )
+        )
+    }
+
+    fun toggleNetworkRequest() {
```
Comment (luizgrp) [inline media-sample/src/main/java/com/google/android/horologist/mediasample/ui/settings/DeveloperOptionsScreenViewModel.kt:140]: all those functions that were copied from SettingsScreenViewModel should be removed from there as they became unused?

I think SettingsScreenViewModel will be empty as a result of that - which is fine, we will be adding some code into it in the future

Comment (laiyichin) [reply media-sample/src/main/java/com/google/android/horologist/mediasample/ui/settings/DeveloperOptionsScreenViewModel.kt:140]: As discussed offline, we will do cleaning up with a follow-up CL if needed.

**Pull request merged**